plugins/out_kinesis_firehose/firehose_api.c

```diff
@@ -216,14 +217,26 @@ static int process_event(struct flb_firehose *ctx, struct flush *buf,
                          ctx->delivery_stream);
             return 2;
         }
-        /* guess space needed to write time_key */
-        len = 6 + strlen(ctx->time_key) + 6 * strlen(ctx->time_key_format);
+
+        /* format time output and return the length */
+        len = flb_aws_strftime_precision(&out_buf, ctx->time_key_format, tms);
```

Comment:
Handle case where `len = -1` because of failure? IIRC we discussed this case last week. Failure of time_key should just skip adding the time_key and not lead to failure of full record.

Reply:
Sorry I see this is down below